aria_data_loaders/aria_data_utils/spot_aria_p1_node.py

```diff
@@ -0,0 +1,591 @@
+        self.spotWorld_T_aria = None
+        self.spotWorld_T_aria_pose_of_interest = None
+        self._act_flag = False
+
+        # Initialize subscribers for listening to aria w.r.t ariaWorld transform and aria pose of interest w.r.t ariaWorld transform
+        self.ariaWorld_T_aria_subscriber = rospy.Subscriber(
+            "/aria_current_pose", PoseStamped, self.ariaWorld_T_aria_callback
+        )
+        self.ariaWorld_T_aria_pose_of_interest_subscriber = rospy.Subscriber(
+            "/aria_pose_of_interest",
+            PoseStamped,
+            self.ariaWorld_T_aria_pose_of_interest_callback,
+        )
+        self._demo_trigger_sub = rospy.Subscriber(
+            "/demo_trigger", Bool, self.demo_trigger_callback
+        )
+
+        # Run in a loop until both poses are received
+        while not rospy.is_shutdown() and (
+            self.spotWorld_T_aria is None
+            or self.spotWorld_T_aria_pose_of_interest is None
+        ):
+            if self.spotWorld_T_aria is None:
+                rospy.logwarn_throttle(
+                    5.0, "Waiting for transform from aria to ariaWorld"
+                )
+            if self.spotWorld_T_aria_pose_of_interest is None:
+                rospy.logwarn_throttle(
+                    5.0, "Waiting for transform from aria pose of interest to ariaWorld"
+                )
+            rospy.sleep(1.0)
+
+        while not rospy.is_shutdown() and not self._act_flag:
+            rospy.logwarn_throttle(5.0, "Waiting for demo trigger")
+            rospy.sleep(1.0)
+
+        while not rospy.is_shutdown():
+            if (
+                self.spotWorld_T_aria is not None
+                and self.spotWorld_T_aria_pose_of_interest is not None
+                and self._act_flag
+            ):
+                print("Got both poses Will Start Nav-Pick-Nav-Place")
+
+                # Nav to Pose of interest (Aria's cpf frame when it saw the object)
+                nav_loc_for_pick = self.goto_nav_pose_for_pick
+                rospy.loginfo(f"Nav loc for pick: {nav_loc_for_pick}")
+
+                # Pick the object
+                pick_object = "bottle"
+
+                # Nav to Pose of interest (Aria wearer's last location)
+                nav_loc_for_place = self.goto_nav_pose_for_place
+                rospy.loginfo(f"Nav loc for place: {nav_loc_for_place}")
+                # Place the object
+                place_loc = self.get_place_pose_to_wearer(self.spotWorld_T_aria)
+
+                # Run Nav-Pick-Nav-Place
+                self.run_nav_pick_nav_place(
+                    nav_loc_for_pick, pick_object, nav_loc_for_place, place_loc
+                )
+
+                input("Press Enter to continue...")
+
+            else:
+                rospy.WARN(
+                    f"Waiting for {self.spotWorld_T_aria} and {self.spotWorld_T_aria_pose_of_interest}"
+                )
+
+    def demo_trigger_callback(self, msg):
```

Comment:
rename `demo_trigger` to `spot_fetch_and_handoff_trigger`, similar renaming for the flag

Reply:
updated